mautic/mautic
# Sms send by segments

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/346
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This is huge update to text messages. 
PR added segments to SMS bundle as broadcast. 
Same approach like emails. 

- [x] Segments support for SMS in UX similar like in emails
- [x] Sent SMS by broadcast command 
`php bin/console mautic:broadcast:send --channel=sms --batch=100` 

![image](https://user-images.githubusercontent.com/462477/65414162-67379c80-ddf3-11e9-93f0-20ad5dd12f87.png)

- [x] Do Not Contact for channels in Contact list  view
![image](https://user-images.githubusercontent.com/462477/67222395-88f56500-f42d-11e9-8a44-c81a348761d0.png)


- [x] Do Not Contact for channels in Contact detail view
<img width="400px" src="https://user-images.githubusercontent.com/462477/67222485-b80bd680-f42d-11e9-8e7e-56d9e68c2811.png" >

- [x] Sent/failed logging
that means - new is_failed columns and details column in sms_message_stats table
![image](https://user-images.githubusercontent.com/462477/67222786-526c1a00-f42e-11e9-94c9-f1d9e33ed8ce.png)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Setup SMS transport
3. Try create SMS
4. Now you can choose between segments/campaign SMS type
5. Create campaign SMS type and test as usual
6. Create segments SMS type and setup publish from/up date
7. Then wait to `mautic:broadcast:sent` cron or try `php bin/console mautic:broadcast:send --channel=sms`
8. You should receive SMS to your list  of contacts from selected segments
